### ISSUE TITLE
Template function empty returns object reference not set exception

### ISSUE THREAD

**Issue by BernieWhite** [Azure/PSRule.Rules.Azure#1025]
**Description of the issue**

When an string pass from a parameter is tested with `empty()` it may return `false` even when an empty string is provided as the parameter value.

This is because the `empty` function is not detecting and converting `JValue` types to determine if the string is empty.

**Expected behaviour**

The `empty` function should automatically perform conversion and return empty if an empty string is provided.

**Error output**

```text
Object reference not set to an instance of an object.
```

**Module in use and version:**

- Module: PSRule.Rules.Azure
- Version: **1.9.0-B2110025**

Captured output from `$PSVersionTable`:

```text
Name                           Value
----                           -----
PSVersion                      7.1.4
PSEdition                      Core
GitCommitId                    7.1.4
OS                             Microsoft Windows 10.0.22000
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```
